tests/test_dump_file.py

```diff
@@ -318,3 +325,320 @@ def test_faulty_dump_schema(tmp_path, schema):
+        assert read_snaps[i].has_typeid()
+        assert numpy.allclose(read_snaps[i].typeid, snaps[i].typeid[order])
+        assert read_snaps[i].has_mass()
+        assert numpy.allclose(read_snaps[i].mass, snaps[i].mass[order])
+        assert read_snaps[i].has_molecule()
+        assert numpy.allclose(read_snaps[i].molecule, snaps[i].molecule[order])
+        assert read_snaps[i].has_charge()
+        assert numpy.allclose(read_snaps[i].charge, snaps[i].charge[order])
+
+
+@pytest.mark.skipif(not has_lammps, reason="lammps not installed")
+def test_copy_from_lammps(snap, tmp_path):
+    # take dump round trip through LAMMPS
+    ref_snap = copy.deepcopy(snap)
+    ref_snap.id = [12, 1, 2]
+    ref_snap.typeid = [2, 1, 2]
+    ref_snap.mass = [3, 2, 3]
+    ref_snap.molecule = [2, 0, 1]
+    ref_snap.charge = [-1, 0, 1]
+
+    snap.id = [1, 2, 12]
+    snap.position = [[0.1, 0.2, 0.3], [-0.4, -0.5, -0.6], [0.7, 0.8, 0.9]]
+
+    # create data and dump file using lammpsio
+    filename_data = tmp_path / "atoms.data"
+    lammpsio.DataFile.create(filename_data, ref_snap)
+    assert filename_data.exists
+
+    filename_dump = tmp_path / "atoms.lammpstrj"
+    schema = {"id": 0, "position": (1, 2, 3)}
+    lammpsio.DumpFile.create(filename_dump, schema, snap)
+    assert filename_dump.exists
+
+    # read dump and data file into LAMMPS
+    lmps = lammps.lammps(cmdargs=["-log", f"{tmp_path}/log.lammps"])
+    cmds = ["atom_style full"]
+    cmds += [f"read_data {filename_data}"]
+    cmds += [f"read_dump {filename_dump} 10 x y z"]
+
+    # write dump file out of LAMMPS
+    cmds += [f"dump mydump all custom 1 {filename_dump} id x y z"]
+    cmds += ["run 0"]
+    lmps.commands_list(cmds)
+    lmps.close()
+
+    f = lammpsio.DumpFile(filename_dump, schema, copy_from=ref_snap)
+    read_snap = [s for s in f][0]
+
+    assert read_snap.N == snap.N
+    assert read_snap.step == snap.step
+    assert numpy.allclose(read_snap.box.low, snap.box.low)
+    assert numpy.allclose(read_snap.box.high, snap.box.high)
+    if snap.box.tilt is not None:
+        assert numpy.allclose(read_snap.box.tilt, snap.box.tilt)
+    else:
+        assert read_snap.box.tilt is None
+    assert read_snap.has_id()
+    assert numpy.allclose(read_snap.id, snap.id)
+    assert read_snap.has_position()
+    assert numpy.allclose(read_snap.position, snap.position)
+    assert not read_snap.has_image()
+    assert not read_snap.has_velocity()
+    assert read_snap.has_typeid()
+    assert numpy.all(read_snap.typeid == [1, 2, 2])
+    assert read_snap.has_mass()
+    assert numpy.allclose(read_snap.mass, [2, 3, 3])
+    assert read_snap.has_molecule()
+    assert numpy.all(read_snap.molecule == [0, 1, 2])
+    assert read_snap.has_charge()
+    assert numpy.allclose(read_snap.charge, [0, 1, -1])
+
+
+@pytest.mark.skipif(not has_lammps, reason="lammps not installed")
+def test_copy_from_topology_lammps(snap_8, tmp_path):
```

Comment:
Same comment here. If we've verified the topology info passes through LAMMPS correctly for the DataFile, I don't see a reason to separately test the `copy_from` feature on its again. If the test is doing something in addition to that, please clarify!